index.js

```diff
@@ -0,0 +1,41 @@
+const {generateSW, injectManifest} = require('workbox-build');
+
+function handleError(msg, logger) {
+    logger.error(msg);
+    return Promise.resolve({
+        status: 'error',
+        message: msg
+    });
+}
+
+function handleSuccess({count, size}, logger, swDest) {
+    const message = `Generated ${swDest} which will precache ${count} files, or ${size} bytes.`;
+    logger.info(message);
+    return {
+        status: 'success',
+        message: message
+    }
+}
+
+function run(config, {logger}) {
+    if(Object.keys(config).length === 0) {
+        return handleError('No configuration provided for Workbox plugin.', logger);
+    } else {
+        switch(config.method) {
+            case 'generateSW':
+                return generateSW(config.workboxConfig).then(resp => handleSuccess(resp, logger, config.workboxConfig.swDest));
+            case 'injectManifest':
+                return injectManifest(config.workboxConfig).then(resp => handleSuccess(resp, logger, config.workboxConfig.swDest));
```

Comment:
Instead of a switch statement that you have to keep adding to, could you just maintain an array of valid method names that you check against? And if you import the workbox methods into an object (`workbox`, for example), you could dynamically call the specified workbox method with `workbox[config.method](config.workboxConfig).then(....)`

Reply:
Actually, you don't even need to maintain an array of methods. You could just check for the existence of a method name as a property of the imported `workbox` object.